labs/a10.md

```diff
@@ -0,0 +1,371 @@
+---
+title: Lab 10 - Special Topics
+layout: lab
+---
+
+## Docker
+This exercise is designed to give you some hands-on experience with Docker!
+By the end of this assignment, you should be able to:
+- Create and use a Docker container interactively
+- Create a Dockerfile, which allows you to declaratively define your containers
+- Run detached containers and understand port forwarding
+- Use `docker-compose` to run a multi-container web application
+
+Just a forewarning: this lab holds your hand until the last section. Not that the last part is super hard, but it'll
+have a lot less instruction than previous portions which are designed to gently introduce you to Docker.
+
+[Here is a link to the Google form](https://goo.gl/forms/BJ5f7OQyCdjCfIzm1)
+
+### Installing Docker
+Unfortunately, Docker isn't in the official Debian `apt` repository. This means there are two methods we can use to
+install it:
+
+1. Add the official Docker repository to `/etc/apt/sources.list`. This allows us to update the version of Docker we
+have on our machines with a simple `apt update` and `apt upgrade`
+2. Download the `.deb` manually and just install it. If we want to upgrade Docker after we do this, we'll have to do it
+manually as well.
+
+You're welcome to use either method to install Docker. For the first method, refer to the [Docker website for
+instructions](https://docs.docker.com/install/linux/docker-ce/debian/#install-docker-ce). For the second method, here
+is a link to a recent `.deb` you can use:
+`https://download.docker.com/linux/debian/dists/stretch/pool/stable/amd64/docker-ce_18.03.0~ce-0~debian_amd64.deb`. At
+this point in the course, I trust you can figure out how to install it just given that link 🙂 _hint: use `wget` and`dpkg`_
```

Comment:
This emoji doesn't show up on Linux, at least not for me due to lack of fonts, but it's probably ok, it should work on the OCF machines and on most people's computers I imagine